cli/src/main/java/bisq/cli/NegativeNumberOptions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.cli;
+
+import java.math.BigDecimal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static java.util.Arrays.stream;
+
+class NegativeNumberOptions {
+
+    private final Map<Integer, String> negativeNumberParams = new HashMap<>();
+
+    String[] removeNegativeNumberOptions(String[] args) {
+        // Cache any negative number params that will be rejected by the parser.
+        // This should be called before command line parsing.
+        for (int i = 0; i < args.length; i++) {
+            if (isNegativeNumber.test(args[i])) {
+                String param = args[i];
+                negativeNumberParams.put(i - 1, new BigDecimal(param).toString());
```

Comment:
I use `negativeNumberParams.put(i - 1, ...)` because `i-1` is the index of the api method param, but `args` includes the method name too.
   ```
    args[0] = method name
    negativeNumberParams.get(0) = 1st parameter of method name ( i.e., args[1] )
```
I added an explanation in commit e09b821

Reply:
Wouldn't it be more natural to start the for loop at 1 then? Like `for (int i = 1; i < args.length; i++) {` since it's known that argument 0 is irrelevant and it would also make the code self explanatory.

Reply:
Yes -- commit  723fc8f.